crates/red_knot/src/lib.rs

```diff
@@ -107,3 +120,62 @@ impl std::fmt::Display for Name {
         f.write_str(self.as_str())
     }
 }
+
+// NOTE: modeled after red_knot::files::FilesInner; not wrapped with Arc(s) for now, because it's
+// not clear that's necessary
+#[derive(Debug, Default)]
+pub struct Names {
```

Comment:
I think I would move this to the symbol table since it is the only place where this is used (and may also not require to be public)

Reply:
I agree, I'd just put this in `symbols.rs` for now. If we do want to split it out, I'd rather have it in its own module than directly in `lib.rs`.

Reply:
I put the implementation in `lib.rs` because that's where `Name` is defined. Following the example of `FileId` and `Files`, it might make sense for `Names` to get its own `.rs` module. If we follow @MichaReiser's suggestion to intern earlier than at the level of a module's symbol table, then it definitely wouldn't belong in `symbols.rs`.